mat64/io_test.go

```diff
@@ -39,3 +52,93 @@ func TestDenseRW(t *testing.T) {
 		}
 	}
 }
+
+func TestDenseRWNaNInf(t *testing.T) {
+	d := NewDense(1, 4, []float64{0, math.Inf(-1), math.Inf(+1), math.NaN()})
+	buf, err := d.MarshalBinary()
+	if err != nil {
+		t.Errorf("error encoding: %v\n", err)
+	}
+
+	var got Dense
+	err = got.UnmarshalBinary(buf)
+	if err != nil {
+		t.Errorf("error decoding: %v\n", err)
+	}
+
+	for i, v := range []bool{
+		got.At(0, 0) == 0,
+		math.IsInf(got.At(0, 1), -1),
+		math.IsInf(got.At(0, 2), +1),
+		math.IsNaN(got.At(0, 3)),
+	} {
+		if !v {
+			t.Errorf("r/w test d.At(0,%d) failed\nwant=%#v\n got=%#v\n", i, d.At(0, i), got.At(0, i))
+		}
+	}
+}
+
+func TestVectorRW(t *testing.T) {
+	for i, test := range []*Vector{
+		NewVector(0, []float64{}),
+		NewVector(4, []float64{1, 2, 3, 4}),
+		NewVector(6, []float64{1, 2, 3, 4, 5, 6}),
+		NewVector(9, []float64{1, 2, 3, 4, 5, 6, 7, 8, 9}),
+		NewVector(9, []float64{1, 2, 3, 4, 5, 6, 7, 8, 9}).ViewVec(0, 3),
+		NewVector(9, []float64{1, 2, 3, 4, 5, 6, 7, 8, 9}).ViewVec(1, 3),
+		NewVector(9, []float64{1, 2, 3, 4, 5, 6, 7, 8, 9}).ViewVec(0, 8),
+		&Vector{
+			mat: blas64.Vector{
+				Data: []float64{0, 1, 2, 3, 4, 5, 6},
+				Inc:  3,
+			},
+			n: 3,
+		},
+	} {
+		buf, err := test.MarshalBinary()
+		if err != nil {
+			t.Errorf("error encoding test #%d: %v\n", i, err)
+		}
+
+		nrows, ncols := test.Dims()
+		sz := nrows*ncols*sizeFloat64 + sizeInt64
+		if len(buf) != sz {
+			t.Errorf("encoded size test #%d: want=%d got=%d\n", i, sz, len(buf))
+		}
+
+		var got Vector
+		err = got.UnmarshalBinary(buf)
+		if err != nil {
+			t.Errorf("error decoding test #%d: %v\n", i, err)
+		}
+
+		if !Equal(&got, test) {
+			t.Errorf("r/w test #%d failed\nwant=%#v\n got=%#v\n", i, test, &got)
+		}
+	}
+}
+
+func TestVectorRWNaNInf(t *testing.T) {
```

Comment:
Can't these NaN and Inf tests be made just a test case within the main test routines?